src/preview/topics.rs

```diff
@@ -0,0 +1,292 @@
+use momento_protos::cache_client::pubsub::{
+    self, pubsub_client::PubsubClient, PublishRequest, SubscriptionRequest,
+};
+use tonic::{codegen::InterceptedService, transport::Channel};
+
+use crate::MomentoError;
+use crate::{
+    endpoint_resolver::MomentoEndpointsResolver,
+    grpc::header_interceptor::HeaderInterceptor,
+    utils::{connect_channel_lazily, user_agent},
+    MomentoResult,
+};
+
+type ChannelType = InterceptedService<Channel, HeaderInterceptor>;
+
+pub struct Pubsub {
+    client: pubsub::pubsub_client::PubsubClient<ChannelType>,
+}
+
+/// Work with topics, publishing and subscribing.
+impl Pubsub {
+    pub fn connect(
+        auth_token: String,
+        momento_endpoint: Option<String>,
+        user_application_name: Option<&str>,
+    ) -> MomentoResult<Self> {
+        let momento_endpoints = MomentoEndpointsResolver::resolve(&auth_token, momento_endpoint)?;
+        let channel = connect_channel_lazily(&momento_endpoints.data_endpoint.url)?;
+        let authorized_channel = InterceptedService::new(
+            channel,
+            HeaderInterceptor::new(
+                &auth_token,
+                &user_agent(user_application_name.unwrap_or("sdk")),
+            ),
+        );
+        Ok(Self {
+            client: pubsub::pubsub_client::PubsubClient::new(authorized_channel),
+        })
+    }
+
+    /// Publish a value to a topic.
+    /// The cache is used as a namespace for your topics, and it needs to exist.
+    /// You don't create topics, you just start using them.
+    pub async fn publish(
+        &self,
```

Comment:
oh, right, so this is the thing that makes it instance method as opposed to static method?

Reply:
yeah `&self` says "this function receives the self instance _as an immutable reference_"